### PR TITLE
buddy request fix for modified brach

### PR DESCRIPTION
**Pull request by sathishavm**
* Trigger a buddy request for the modified branch within the buddy grace_period.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: https://zendesk.atlassian.net/browse/BRE-971

### Risks
- Level: Low

### CODE REVIEW

File: app/models/deploy_service.rb [hunk 1]
```diff
@@ -78,6 +78,7 @@ def latest_approved_deploy(reference, project)
 
   def copy_approval_from_last_deploy(deploy)
     return false unless last_deploy = latest_approved_deploy(deploy.reference, deploy.stage.project)
+    return false if deploy.changeset_to(last_deploy).commits.any?
```
Comment (grosser) [inline app/models/deploy_service.rb]: might be cheaper to resolve commit and check if it is not the same ... also make sure changeset_to does not compare refs which would be the same

File: test/models/deploy_service_test.rb [hunk 2]
```diff
@@ -85,6 +86,15 @@ def create_previous_deploy(ref, stage, successful: true, bypassed: false)
         end
       end
 
+      describe "similar deploy reference with different commit sha" do
```
Comment (grosser) [inline test/models/deploy_service_test.rb]: similar -> same

File: test/models/deploy_service_test.rb [hunk 3]
```diff
@@ -33,8 +33,9 @@
       before { BuddyCheck.stubs(:enabled?).returns(true) }
       let(:deploy) { deploys(:succeeded_production_test) }
 
-      def create_previous_deploy(ref, stage, successful: true, bypassed: false)
-        job = project.jobs.create!(user: user, command: "foo", status: successful ? "succeeded" : 'failed')
+      def create_previous_deploy(ref, stage, successful: true, bypassed: false, commit: '')
```
Comment (grosser) [inline test/models/deploy_service_test.rb]: empty commit seems kinda strange ... maybe nil ?

File: app/models/deploy_service.rb (hunk 1 above)
Comment (sathishavm) [reply app/models/deploy_service.rb]: We may cache the changeset in the previous Review step, it should re-use the cache.

**Pull request merged**